route/build/src/datetime_data.f90

```diff
@@ -443,4 +465,35 @@ logical(lgt) FUNCTION fn_is_le(this, that)
     end if
   END FUNCTION fn_is_le
 
+  real(dp) FUNCTION fn_delta(this, that) result(ans)
+    ! this - that give in time difference in second
+    implicit none
+    class(datetime),     intent(in)  :: this
+    class(datetime),     intent(in)  :: that
+    ! local variables
+    real(dp)                         :: sec_in_day1
+    real(dp)                         :: sec_in_day2
+    real(dp)                         :: diffDay
+    integer(i4b)                     :: ierr
+    type(datetime)                   :: date1
+    type(datetime)                   :: date2
+    real(dp)                         :: julday1
+    real(dp)                         :: julday2
+    character(len=strLen)            :: cmessage
+
+    sec_in_day1 = real(this%ih*3600+this%imin*60, kind=dp) +this%dsec
```

Comment:
3600 could be a parameter for seconds in an hour.

Reply:
replaced 3600 and 60 with `secprday` and `secprmin`, respectively